nimbus-ui/nimbusui/src/app/components/platform/form/elements/rich-text.component.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * @license
+ * Copyright 2016-2018 the original author or authors.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+import { NgModel, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Component, ViewChild, forwardRef, ChangeDetectorRef } from '@angular/core';
+import { WebContentSvc } from '../../../../services/content-management.service';
+import { BaseControl } from './base-control.component';
+import { ControlSubscribers } from './../../../../services/control-subscribers.service';
+
+export const CUSTOM_INPUT_CONTROL_VALUE_ACCESSOR: any = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => RichText),
+    multi: true
+};
+
+/**
+ * \@author Tony Lopez
+ * \@whatItDoes RichText is rich text editor component based on Quill.
+ * \@howToUse <nm-input-rich-text></nm-input-rich-text>
+ * 
+ */
+@Component({
+    selector: 'nm-input-rich-text',
+    providers: [CUSTOM_INPUT_CONTROL_VALUE_ACCESSOR, WebContentSvc, ControlSubscribers],
+    template: `
+    <nm-input-label
+        [element]="element" 
+        [for]="element.config.code"
+        [required]="requiredCss">
+    </nm-input-label>
+
+    <p-editor
```

Comment:
We would need to address the visibility condition for the component.

Reply:
Visibility attribute is addressed in `form-element.component.ts`, it's only parent component.

It wouldn't hurt to wrap it with a visibility check but I thought it might not be necessary. Thoughts?

Reply:
If it is already addressed in the parent component, then we should be good.